odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,331 @@
+
+
+__all__ = ('AttenuatedRayTransform', 'AttenuatedRayBackprojection',
+           'NIFTYREC_AVAILABLE')
+_SUPPORTED_IMPL = ('niftyrec', 'niftyrec_gpu')
+
+
+class AttenuatedRayTransform(Operator):
+
+    """The discrete attenuated Ray transform between L^p spaces."""
+
+    def __init__(self, dom, geometry, attenuation=None, psf=None,
+                 impl='niftyrec'):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Discretized space, a 3 dimensional domain of the forward projector
+        geometry : `ParallelHoleCollimatorGeometry`
+            The geometry of the transform. An ODL geometry instance
+            that contains information about the operator range
+        attenuation : domain `element-like` structure of the shape
+            `dom.shape`, optional
+            Linear attenuation map for SPECT.
+        psf : `element-like` structure, optional
+            Point Spread Functions for the correction of the
+            collimator-detector response
+        impl : {'niftyrec', 'niftyrec_gpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+
+            'niftyrec' : `NiftyRec`_ tomography toolbox using the `NiftyPy`_
+            Python wrapper and CPU
+
+            'niftyrec_gpu' : `NiftyRec`_ tomography toolbox using the
+            `NiftyPy`_ Python wrapper and GPU
+
+        References
+        ----------
+        .. _NiftyRec: http://niftyrec.scienceontheweb.net/wordpress/
+
+        .. _NiftyPy: https://github.com/spedemon/niftypy
+
+        """
+        if not isinstance(dom, DiscreteLp):
+            raise TypeError('discretized domain {!r} is not a `DiscreteLp`'
+                            ' instance.'.format(dom))
+
+        if not isinstance(geometry, ParallelHoleCollimatorGeometry):
+            raise TypeError('geometry {!r} is not a'
+                            '`ParallelHoleCollimatorGeometry` instance.'
+                            ''.format(geometry))
+
+        ran = uniform_discr_frompartition(geometry.partition)
+        super().__init__(dom, ran, linear=True)
+
+        impl, impl_in = str(impl).lower(), impl
+        if impl not in _SUPPORTED_IMPL:
+            raise ValueError('implementation {!r} not supported.'
+                             ''.format(impl_in))
+
+        if impl.startswith('niftyrec'):
+            if not NIFTYREC_AVAILABLE:
+                raise ValueError('NiftyRec back-end not available.')
+            if impl == 'niftyrec_gpu' and dom.shape != (128, 128, 128):
+                raise NotImplementedError('NiftyRec GPU back-end only '
+                                          'supports volume size '
+                                          '(128, 128, 128), got {}'
+                                          ''.format(dom.shape))
+            if impl == 'niftyrec_gpu':
+                self._use_gpu = (impl == 'niftyrec_gpu')
```

Comment:
Change this if ... else ... statement to

`self._use_gpu = (impl == 'niftyrec_gpu')`